annotation-processing/src/main/java/com/exactpro/epfast/annotation/processing/FastProcessor.java

```diff
@@ -9,122 +12,306 @@
+            element));
     }
 
-    private HashMap<String, Element> filterDuplicates(HashMap<String, ArrayList<Element>> nameClassMap) {
-        HashMap<String, Element> filteredNameClassMap = new HashMap<>();
-        for (Map.Entry<String, ArrayList<Element>> entry : nameClassMap.entrySet()) {
-            String fastTypeName = entry.getKey();
-            ArrayList<Element> elements = entry.getValue();
-            Element element = getSingleElementOrNull(elements);
-            if (element == null) {
-                reportDuplicates(fastTypeName, elements);
+    private void organizeInPackages(HashMap<String, ArrayList<Element>> fastTypesByName) {
+        HashMap<String, FastPackageElement> noPackageFastTypes = new HashMap<>();
+        for (Map.Entry<String, ArrayList<Element>> entry : fastTypesByName.entrySet()) {
+            HashSet<String> packagesForFastTypeElements = new HashSet<>();
+            for (Element element : entry.getValue()) {
+                String fastPackage = getFastPackageOf(element);
+                if (fastPackage != null) {
+                    if (packagesForFastTypeElements.contains(fastPackage)) {
+                        reportDuplicates(entry.getKey(), entry.getValue());
+                    } else {
+                        FastPackageElement packageElement = fastPackageFactory.getFastPackage(fastPackage);
+                        //if packageElement is null, then this package was duplicated and reported already.
+                        if (packageElement == null) {
+                            continue;
+                        }
+                        packagesForFastTypeElements.add(fastPackage);
+                        packageElement.addElement(entry.getKey(), element);
+                        fastPackageFactory.addPackageElement(packageElement);
+                    }
+                } else {
+                    String javaPackage = elementUtils.getPackageOf(element).toString();
+                    if (packagesForFastTypeElements.contains(javaPackage)) {
+                        reportDuplicates(entry.getKey(), entry.getValue());
+                    } else {
+                        packagesForFastTypeElements.add(javaPackage);
+                        noPackageFastTypes.computeIfAbsent(javaPackage, key -> new FastPackageElement(javaPackage))
+                            .addElement(entry.getKey(), element);
+                    }
+                }
+
+            }
+        }
+        String rootPackage = getRootPackage(noPackageFastTypes.values().stream()
+            .map(FastPackageElement::getName)
+            .collect(Collectors.toList()));
+        fastPackageFactory.setRootPackage(rootPackage);
+        noPackageFastTypes.putIfAbsent(rootPackage, new FastPackageElement(rootPackage));
+        for (Map.Entry<String, FastPackageElement> entry : noPackageFastTypes.entrySet()) {
```

Comment:
All @FastType annotated which don't belong to explicitly annotated Fast package should go to root package